src/Compilers/Lua/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -307,6 +316,16 @@ public static SyntaxToken Token(SyntaxTriviaList leading, SyntaxKind kind, strin
         public static SyntaxToken Literal(SyntaxTriviaList leading, string text, ulong value, SyntaxTriviaList trailing) =>
             new(InternalSyntax.SyntaxFactory.Literal(leading.Node, text, value, trailing.Node));
 
+        /// <summary>
+        /// Creates a token with kind NumericLiteralToken from the text and corresponding complex value.
+        /// </summary>
+        /// <param name="leading">A list of trivia immediately preceding the token.</param>
+        /// <param name="text">The raw text of the literal.</param>
+        /// <param name="value">The complex value to be represented by the returned token.</param>
+        /// <param name="trailing">A list of trivia immediately following the token.</param>
+        public static SyntaxToken Literal(SyntaxTriviaList leading, string text, Complex value, SyntaxTriviaList trailing) =>
+            new(InternalSyntax.SyntaxFactory.Literal(leading.Node, text, value, trailing.Node));
+
```

Comment:
We need something here to validate that the `Complex` number doesn't have a [real](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.complex.real?view=net-6.0#system-numerics-complex-real) part